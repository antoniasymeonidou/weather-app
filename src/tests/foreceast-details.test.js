import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../components/forecast-details";

describe("ForecastDetails", () => {
    const mockForecast = {
      date: 123,
      temperature: {
        max: 11,
        min: 4,
      },
      humidity: 30,
      wind: {
        speed: 13,
      },
    };

    it("renders correctly", () => {
        const { asFragment } = render(<ForecastDetails forecast={mockForecast} />);
        expect(asFragment()).toMatchSnapshot();
      });
      it("renders the correct amount of props", () => {
        const { getByText } = render(<ForecastDetails forecast={mockForecast} />);
    
        expect(getByText("Monday 30th Apr")).toHaveClass("detailed-date");
        expect(getByText("Max Temperature: 11°c")).toHaveClass("max-temp");
        expect(getByText("Min Temperature: 4°c")).toHaveClass("min-temp");
        expect(getByText("Humidity: 30%")).toHaveClass("detailed-humidity");
        expect(getByText("Wind: 13mph")).toHaveClass("detailed-wind");
      });
    });




