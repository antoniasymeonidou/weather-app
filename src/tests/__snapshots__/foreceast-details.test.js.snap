// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <h3
      class="detailed-date"
    >
      Monday 30th Apr
    </h3>
    <div
      class="detailed-temperature"
    >
      <div
        class="max-temp"
      >
        Max Temperature: 11°c
      </div>
      <div
        class="min-temp"
      >
        Min Temperature: 4°c
      </div>
    </div>
    <div
      class="detailed-humidity"
    >
      Humidity: 30%
    </div>
    <div
      class="detailed-wind"
    >
      Wind: 13mph
    </div>
  </div>
</DocumentFragment>
`;
